### ISSUE TITLE
PatientSelectComponent bugs out when closed

### ISSUE THREAD

**Issue by sworup** [openmsupply/mobile#4838]
## Describe the bug

- PatientSelectComponent bugs out when closed with a warning saying:

```
ERROR  Warning: Can't perform a React state update on an unmounted component. This is a no-op, but it indicates a memory leak in your application. To fix, cancel all subscriptions and asynchronous tasks in a useEffect cleanup function.
    in PatientSelectComponent (created by Connect(PatientSelectComponent))
    in Connect(PatientSelectComponent) (at SceneView.tsx:122)
    in StaticContainer
    in EnsureSingleNavigator (at SceneView.tsx:114)
    in SceneView (at useDescriptors.tsx:153)
    in RCTView (at View.js:34)
    in View (at BottomTabView.tsx:55)
    in SceneContent (at BottomTabView.tsx:172)
    in RNSScreen (at createAnimatedComponent.js:217)
    in AnimatedComponent (at createAnimatedComponent.js:278)
    in AnimatedComponentWrapper (at src/index.native.tsx:127)
    in Screen (at ResourceSavingScene.tsx:30)
    in ResourceSavingScene (at BottomTabView.tsx:166)
    in RNSScreenContainer (at src/index.native.tsx:165)
    in ScreenContainer (at BottomTabView.tsx:146)
    in SafeAreaProviderCompat (at BottomTabView.tsx:145)
    in BottomTabView (at createBottomTabNavigator.tsx:45)
    in BottomTabNavigator (at TabNavigator.js:27)
    in TabNavigator (at Wizard.js:44)
    in RCTView (at View.js:34)
    in View (at DataTablePageView.js:36)
    in RCTView (at View.js:34)
    in View (at DataTablePageView.js:35)
    in _c
    in RCTView (at View.js:34)
    in View (at Wizard.js:32)
    in WizardComponent (created by Connect(WizardComponent))
    in Connect(WizardComponent) (at VaccineDispensingPage.js:36)
    in VaccineDispensingPageComponent (created by Connect(VaccineDispensingPageComponent))
    in Connect(VaccineDispensingPageComponent) (at SceneView.tsx:122)
    in StaticContainer
    in EnsureSingleNavigator (at SceneView.tsx:114)
    in SceneView (at useDescriptors.tsx:153)
    in RCTView (at View.js:34)
    in View (at CardContainer.tsx:245)
    in RCTView (at View.js:34)
    in View (at CardContainer.tsx:244)
    in RCTView (at View.js:34)
    in View (at CardSheet.tsx:33)
    in CardSheet (at Card.tsx:573)
    in RCTView (at View.js:34)
    in View (at createAnimatedComponent.js:217)
    in AnimatedComponent (at createAnimatedComponent.js:278)
    in AnimatedComponentWrapper (at Card.tsx:555)
    in PanGestureHandler (at GestureHandlerNative.tsx:13)
    in PanGestureHandler (at Card.tsx:549)
    in RCTView (at View.js:34)
    in View (at createAnimatedComponent.js:217)
    in AnimatedComponent (at createAnimatedComponent.js:278)
    in AnimatedComponentWrapper (at Card.tsx:544)
    in RCTView (at View.js:34)
    in View (at Card.tsx:538)
    in Card (at CardContainer.tsx:206)
    in CardContainer (at CardStack.tsx:623)
    in RNSScreen (at createAnimatedComponent.js:217)
    in AnimatedComponent (at createAnimatedComponent.js:278)
    in AnimatedComponentWrapper (at src/index.native.tsx:127)
    in Screen (at Screens.tsx:74)
    in MaybeScreen (at CardStack.tsx:616)
```

### To reproduce

Steps to reproduce the behaviour:

https://user-images.githubusercontent.com/683452/182303107-272af721-232a-4478-a9cf-5e7b939ece9d.mp4

### Expected behaviour

- When the component unmounts. The component should not trigger state update.

### Agreed Solution

- ```js
    // mobile/src/widgets/FormControl.js
    React.useEffect(() => {
    // Only update state if component comes in focus
    if (isFocused) {
      onInitialiseForm();
      setRefs({ length: inputConfig.length });
    }
    }, [isFocused]);
    ```

### Version and device info

- App version: 8.5.X 
- Tablet model:
- OS version:

### Additional context

This I think is being caused after Kiribati vax changes. So it effects Kiribati Vax for now.

**Comment by sworup** [comment]
## Source of error

This issue comes from this section of code:

https://github.com/openmsupply/mobile/blob/80824641c7e16273c75d5750d70f2e13d5e02de9/src/hooks/useLocalAndRemotePatients.js#L217-L223

And this section of code:
https://github.com/openmsupply/mobile/blob/80824641c7e16273c75d5750d70f2e13d5e02de9/src/widgets/FormControl.js#L80-L85

This error occurs when these two code try to run together.

## Reason

Here throttled is used to trigger local patient search on certain interval, 500 milliseconds, while search keywords are being entered. Since we do not want the search to trigger on every keyword type, we trigger it in half a sec regardless of how fast the text is being typed. Basically same idea as debounce. 

So, throttled is waiting to trigger the method for half a second. The method would be triggered after half a second of being called from code.

The FormControl focus is using `useIsFocused` from react navigation, which triggers every time the windows comes in focus and goes out of focus. It returns a boolean which sets to true if the window is in focus and false if the windows goes out of focus.

Now the above code checks for isFocused and triggers the `onInitialiseForm` method which in turn calls `throttledGetLocalPatients`. Since we are not checking whether isFocused is true or not, the throttledGetLocalPatients method gets called in both cases where the window comes in focus and goes out of focus. 

When the window comes in focus we want to query for patients and it is desired effect so fine.

When the window goes out of focus, if the throttledGetLocalPatients method triggers, the component will already be gone but the throttledGetLocalPatients method triggers half a second later, it fetches the data and tries to update state of component that no longer exist. Hence causing the error.

The reason it says PatientSelectComponent in error is because even though onInitialiseForm is being called from FormControl, which is that search input box. It is redux and it dispatches an event, the event changes states and the updated state is used in PatientSelectComponent.

## Solution

```
React.useEffect(() => {
    if (isFocused) {
      onInitialiseForm();
      setRefs({ length: inputConfig.length });
    }
  }, [isFocused]);
```

🤦🏼

**Comment by sworup** [comment]
Merged with #4846

**Comment by nisha-dangol** [comment]
Moving this to Done since I do not see any app crashes here, esp. with  the back `<` button in Vaccination
